test/unit/gateways/global_collect_test.rb

```diff
@@ -71,6 +86,30 @@ def test_successful_purchase_with_requires_approval_true
     end.respond_with(successful_authorize_response, successful_capture_response)
   end
 
+  def test_purchase_request_with_google_pay
+    stub_comms do
+      @gateway.purchase(@accepted_amount, @google_pay_network_token)
+    end.check_request(skip_response: true) do |_endpoint, data, _headers|
+      assert_equal '320', JSON.parse(data)['mobilePaymentMethodSpecificInput']['paymentProductId'] if JSON.parse(data)['mobilePaymentMethodSpecificInput']
```

Comment:
❓  Why that if ?? => if the test fails because there is not mobilePaymentMethodSpecificInput that's part of test check.

Reply:
This is to check if the value is present and validate if the test was made with google_pay because '320' is the value of Google for paymentProductId.

Reply:
Actually I'm not seeing a reason to keep that if, I'll remove it.